src/opt/ImproveMemoryOps.cpp

```diff
@@ -1082,6 +1082,106 @@ static llvm::Constant *lGetOffsetScaleVec(llvm::Value *offsetScale, llvm::Type *
     return llvm::ConstantVector::get(scales);
 }
 
+struct GatherImpInfo {
```

Comment:
Same questions for this commit:
- why not local static?
- why not class?

Reply:
I've moved them to function scope. It was a struct.